File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/directoryservices/adread/vb/adread.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.DirectoryServices

Public Class ADRead
        public shared sub Main()
        if(Environment.GetCommandLineArgs().Length<>2)
            Console.WriteLine("Usage: " & Environment.GetCommandLineArgs()(0) & " <ad_path>")
            Console.WriteLine ()
            Console.WriteLine ("Press Enter to continue...")
            Console.ReadLine()
            exit sub
        end if

        Dim objDirEnt As DirectoryEntry = new DirectoryEntry(Environment.GetCommandLineArgs(1))
        Console.WriteLine("Name            = " & objDirEnt.Name)
        Console.WriteLine("Path            = " & objDirEnt.Path)
        Console.WriteLine("SchemaClassName = " & objDirEnt.SchemaClassName)
        Console.WriteLine("")
        Console.WriteLine("Properties:")

        Dim tab As string = "    "
        Dim Key As String
        Dim objValue as object

        For Each Key In objDirEnt.Properties.PropertyNames
            Console.Write(tab & Key & " = ")
            Console.WriteLine("")
            For Each objValue In objDirEnt.Properties(Key)
                    Console.WriteLine(tab & tab & objValue.ToString())
            next objValue
        Next
    End Sub
End Class

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/services/directoryservices/adread/cs/adread.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.DirectoryServices;

public class ADRead {
    public static void Main(String[] args) {
        if(args.Length!=1) {
            Console.WriteLine("Usage: " + Environment.GetCommandLineArgs()[0] + " <ad_path>");
            Console.WriteLine ();
            Console.WriteLine ("Press Enter to continue...");
            Console.ReadLine();
            return;
        }

        DirectoryEntry objDirEnt=new DirectoryEntry(args[0]);
        Console.WriteLine("Name            = " + objDirEnt.Name);
        Console.WriteLine("Path            = " + objDirEnt.Path);
        Console.WriteLine("SchemaClassName = " + objDirEnt.SchemaClassName);
        Console.WriteLine("");
        Console.WriteLine("Properties:");

        foreach(String Key in objDirEnt.Properties.PropertyNames) {
            Console.Write("\t{0} = ", Key);
            Console.WriteLine("");
            foreach(Object objValue in objDirEnt.Properties[Key]) {
                Console.WriteLine("\t\t{0}", objValue);
            }
        }
    }
}